src/test/java/spoon/test/compilationunit/TestCompilationUnit.java

```diff
@@ -127,4 +133,24 @@ public void testAddDeclaredTypeInCU() throws IOException {
         assertTrue(fullContent.contains("class Baz"));
         assertTrue(fullContent.contains("class Bla"));
     }
+
+    @Test
+    public void testNewlyCreatedCUWouldGetAPartialPosition() throws IOException {
+        // contract: when a type is created, a CU can be created and added as partial position
+        final Launcher launcher = new Launcher();
+        assertTrue(launcher.getFactory().CompilationUnit().getMap().isEmpty());
+
+        CtClass myNewClass = launcher.getFactory().createClass("my.new.MyClass");
+        assertEquals(SourcePosition.NOPOSITION, myNewClass.getPosition());
+
+        CompilationUnit cu = launcher.getFactory().CompilationUnit().getOrCreate(myNewClass);
+
+        assertNotNull(cu);
```

Comment:
//next call of getOrCreate returns existing cu
assertSame(cu, launcher.getFactory().CompilationUnit().getOrCreate(myNewClass))

Reply:
I'll add it too.